android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -2218,20 +2227,68 @@ public static int getBannerHeight(Context context) {
   }
 
   public static int getKeyboardHeight(Context context) {
-    int defaultHeight = (int) context.getResources().getDimension(R.dimen.keyboard_height);
-    SharedPreferences prefs = context.getSharedPreferences(context.getString(R.string.kma_prefs_name), Context.MODE_PRIVATE);
-
     int orientation = getOrientation(context);
+    return getKeyboardHeight(context, orientation); 
+  }
+
+  public static int getKeyboardHeight(Context context, int orientation) {
+    SharedPreferences prefs = context.getSharedPreferences(context.getString(R.string.kma_prefs_name), Context.MODE_PRIVATE);
     if (orientation == Configuration.ORIENTATION_PORTRAIT) {
-      return prefs.getInt(KMManager.KMKey_KeyboardHeightPortrait, defaultHeight);
-    } else if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
-      return prefs.getInt(KMManager.KMKey_KeyboardHeightLandscape, defaultHeight);
+        return prefs.getInt(KMManager.KMKey_KeyboardHeightPortrait, KMManager.KeyboardHeight_Context_Portrait_Default);
+    } else /*(orientation == Configuration.ORIENTATION_LANDSCAPE) */ {
+        return prefs.getInt(KMManager.KMKey_KeyboardHeightLandscape, KMManager.KeyboardHeight_Context_Landscape_Default);
     }
-
-    return defaultHeight;
   }
 
+  /**
+   * Apply the keyboard height for the current orientation and store it in SharedPreferences.
+   * If height is KMManager.DEFAULT_KEYBOARD_HEIGHT, remove the stored height for that orientation.
+   * @param context Context
+   * @param height int - keyboard height in dp
+   */
   public static void applyKeyboardHeight(Context context, int height) {
+    int orientation = getOrientation(context);
+    applyKeyboardHeight(context, height, orientation); 
+  }
+
+  /**
+   * Apply the keyboard height for the current orientation and store it in SharedPreferences.
+   * If height is KMManager.DEFAULT_KEYBOARD_HEIGHT, remove the stored height for that orientation.
+   * @param context Context
+   * @param height int - keyboard height in dp
+   * @param orientation int - Configuration.ORIENTATION_PORTRAIT or Configuration.ORIENTATION_LANDSCAPE
+   */
+  public static void applyKeyboardHeight(Context context, int height, int orientation) {
+    SharedPreferences prefs = context.getSharedPreferences("KMAPreferences", Context.MODE_PRIVATE);
+    SharedPreferences.Editor editor = prefs.edit();
+    int defaultHeightForContext = orientation == Configuration.ORIENTATION_PORTRAIT ? 
+    KeyboardHeight_Context_Portrait_Default: KeyboardHeight_Context_Landscape_Default;
+    if (height == KMManager.KeyboardHeight_Reset) {
+      // Passing 0 will reset the stored height for this orientation.
+      if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
+        editor.remove(KMManager.KMKey_KeyboardHeightLandscape);
+      } else if (orientation == Configuration.ORIENTATION_PORTRAIT) {
+        editor.remove(KMManager.KMKey_KeyboardHeightPortrait);
+      }
+      editor.commit();
```

Comment:
Can this be removed since it's handled by the suggestion you added from Marc (l. 2290)

Reply:
I don't think it can be removed. 2266-2273 handles 0 values on height to reset. 2275-2281 handles extreme values. 2283-2288 handles normal sharePref operations, and the next part handles layout.

Reply:
Oh the extra commit line. Yes, I suppose that is redundant.